Tools/jenkins/build.dockerfile

```diff
@@ -1,16 +1,36 @@
 FROM jangrewe/gitlab-ci-android
 
-ARG emulator_platform=android-26
-ARG system_image="system-images;${emulator_platform};google_apis;x86"
-ARG platform_image="platforms;${emulator_platform}"
-
+################################ Tools Installation ################################
 RUN apt-get update && \
   apt-get install -y --no-install-recommends \
   nodejs=6.11.4~dfsg-1ubuntu1 \
   build-essential \
   && rm -rf /var/lib/apt/lists/*
 
+################################ Gradle Installation ################################
+# Allow the host to use gradle cache, otherwise gradle will always download plugins & artifacts.
+VOLUME ["/root/.gradle/caches/"]
+
+ARG GRADLE_VERSION=4.10
```

Comment:
We've had trouble with gradle version 4.9 and 4.10 in the past executing some of the DSL we have in our gradle files. Generally we use Gradle 4.6

Reply:
That sounds like a problem with the build system of the path not being deterministic. Our current endeavor is to make a deterministic build system. This passes build and testing in CI. If something is broken, it should be visible in build or test.

Reply:
Well actually, i meant there are  gradle incompatibilities when executing other commands in the deploy process when using versions higher than 4.9. For example: https://github.com/gradle/gradle/issues/6396, which means we would need to make sure to get the correct corresponding of the other plugin to prevent it. Thats why i was saying using 4.6 is a safe bet because we know that there are no incompatibilities with that particular version with the tools we use. However, this is not really a big issue, it just means that we need to vet all other tools we are using in conjunction with gradle to ensure that they are interoperable.